webapp/src/components/systemConsole/surveyDateTime/surveyDateTime.tsx

```diff
@@ -1,86 +1,61 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {format, parse} from 'date-fns';
+import {format} from 'date-fns';
+import {formatInTimeZone} from 'date-fns-tz';
 import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import utils from 'utils/utils';
 
 import SurveyDateSelector from 'components/surveyDateSelector/surveyDateSelector';
 import SurveyTimeSelector from 'components/surveyTimeSelector/surveyTimeSelector';
 import type {CustomSettingChildComponentProp} from 'components/systemConsole/index';
 
-import type {DateTimeConfig} from 'types/plugin';
-
 import './style.scss';
-
-const DEFAULT_SURVEY_TIME = '00:00';
+import type {DateTimeConfig} from 'types/plugin';
 
 function SurveyDateTime({id, setSaveNeeded, onChange, config, setInitialSetting}: CustomSettingChildComponentProp) {
-    const [surveyTime, setSurveyTime] = useState<string>(DEFAULT_SURVEY_TIME);
     const [surveyDate, setSurveyDate] = useState<Date>();
 
-    const [surveyDateTime, setSurveyDateTime] = useState<Date>();
+    // sets the date picker and time picker to the values saved in config on load
+    useEffect(() => {
```

Comment:
Doing this in a `useEffect` looks a bit odd since it's something you want run on first load, but I can't think of any real reason we shouldn't do that since this isn't a performance-sensitive area

Reply:
the config object received by plugin is empty on first load, so we need to keep reacting to its change for a while.